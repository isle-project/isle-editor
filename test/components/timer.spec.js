// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Timer from '@isle-project/components/timer/main.js';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';


// TESTS //

describe( '<Timer />', function test() {
	it( 'renders a timer', () => {
		const timer = renderer.create(
			<Timer
				active={true}
				duration={50}
			/>
		).toJSON();
		expect( timer ).toMatchSnapshot();
	});

	it( 'renders a timer (inactive)', () => {
		const timer = renderer.create(
			<Timer
				active={false}
				duration={50}
			/>
		).toJSON();
		expect( timer ).toMatchSnapshot();
	});

	it( 'renders a timer with a custom legend', () => {
		render(
			<Timer
				legend="My timer"
				active={true}
				duration={50}
			/>
		);
		expect( screen.getByText( 'My timer', { exact: false } ) ).toBeInTheDocument();
	});

	it( 'renders a timer with a custom style', () => {
		render(
			<Timer
				legend="My timer"
				active={true}
				duration={50}
				style={{
					width: '100px',
					height: '100px',
					background: '#f00'
				}}
			/>
		);
		const timer = screen.getByText( 'My timer', {
			exact: false
		});
		expect( timer ).toHaveStyle( 'background: #f00' );
	});
});
