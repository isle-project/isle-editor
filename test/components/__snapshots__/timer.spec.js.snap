// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> renders a timer (inactive) 1`] = `
<div
  className="timer-div timer-info"
  style={{}}
>
  00:50
</div>
`;

exports[`<Timer /> renders a timer 1`] = `
<div
  className="timer-div timer-info"
  style={{}}
>
  00:50
</div>
`;
